odxtools/parameters/tablekeyparameter.py

```diff
@@ -19,6 +20,41 @@ def __init__(self, *, odx_id, table_ref, table_snref, table_row_snref, table_row
         self.table_snref = table_snref
         self.table_row_snref = table_row_snref
 
+        if self.table_ref is None and self.table_snref is None and \
+           self.table_row_ref is None and self.table_row_snref is None:
+            raise ValueError("Either a table or a table row must be defined.")
+
+    def _build_odxlinks(self) -> Dict[OdxLinkId, Any]:
+        result = super()._build_odxlinks()
+
+        result[self.odx_id] = self
+
+        return result
+
+    def _resolve_odxlinks(self, odxlinks: OdxLinkDatabase) -> None:
+        super()._resolve_odxlinks(odxlinks)
+
+        if self.table_ref:
+            self._table = odxlinks.resolve(self.table_ref)
+        if self.table_row_ref:
+            self._table_row = odxlinks.resolve(self.table_row_ref)
```

Comment:
resolve does not accept None and return None?

Reply:
well, the purpose of these methods is to resolve the internally contained references of the object, i.e. it *modifies* the object (akin `list.sort()`)

Reply:
I am asking if the "if self.table_row_ref:" is really necessary

Reply:
ah okay, my bad. the signature of this method currently is [`def resolve(self, ref: OdxLinkRef, expected_type: Optional[Type[T]] = None) -> Any`](https://github.com/mercedes-benz/odxtools/blob/main/odxtools/odxlink.py#L180). So it currently does not accept `None`, but you're right: it might avoid a few case distinctions if it did. On the flipside, it might come with a few more implicit assumptions which one must consider while reading code that uses it. I suppose I have to meditate a bit about it before I can form an opinion...